### ISSUE TITLE
With "salesforcedx-vscode-core>Experimental: Deploy Retrieve" TURNED ON, I get "Could not infer a metadata type" error

### ISSUE THREAD

**Issue by dweaver33** [forcedotcom/salesforcedx-vscode#3114]
### Summary

with "salesforcedx-vscode-core>Experimental: Deploy Retrieve" TURNED ON i tried a package.xml retrieval as seen below by right-clicking the manifest, and using "SFDX: Retrieve Source in Manifest from Org" but that gave me this other error: "Could not infer a metadata type"

### Steps To Reproduce:

1.  Ensure "salesforcedx-vscode-core>Experimental: Deploy Retrieve" is on
1.  Right-click the manifest, and use "SFDX: Retrieve Source in Manifest from Org"

### Expected result

Retrieves metadata in package.xml file

### Actual result

Get the following error message:  "Could not infer a metadata type"

### Additional information

![image](https://user-images.githubusercontent.com/6692418/113355206-06e9bb00-9306-11eb-97b0-12c884020669.png)

**VS Code Version**:
1.55.0
Salesforce Extension v51.7.0

**SFDX CLI Version**:
sfdx-cli/7.93.0 darwin-x64 node-v15.12.0

**OS and version**:
MacOS 11.2.3

**Comment by dweaver33** [comment]
Note that disabling the "salesforcedx-vscode-core>Experimental: Deploy Retrieve" fixes the issue.  

I can also fix the issue by deleting the "workSkillRoutings" folder in the "force-app" directory with the Deploy/Retrieve option on.

**Comment by brpowell** [comment]
Thanks for reporting @dweaver33. Looks like `WorkSkillRouting` was a missing type definition under the hood. Should be able to update that and publish a fix soon.

**Comment by IVerkhosh** [comment]
I have the same issue trying to retrieve Wave objects
c:\GIT_REPOS\ENERGY_INSIGHT_SFDX\force-app\main\default\wave\RestAPI.wdash-meta.xml-meta.xml: Could not infer a metadata type
	<types>
		<members>*</members>
		<name>WaveApplication</name>
	</types>
	<types>
		<members>*</members>
		<name>WaveDashboard</name>
	</types>
	<types>
		<members>*</members>
		<name>WaveDataflow</name>
	</types>
	<types>
		<members>*</members>
		<name>WaveDataset</name>
	</types>
	<types>
		<members>*</members>
		<name>WaveLens</name>
	</types>
	<types>
		<members>*</members>
		<name>WaveRecipe</name>
	</types>
	<types>
		<members>*</members>
		<name>WaveTemplateBundle</name>
	</types>
	<types>
		<members>*</members>
		<name>WaveXmd</name>
	</types>

**Comment by dweaver33** [comment]
@IVerkhosh, see my response [here](https://github.com/forcedotcom/salesforcedx-vscode/issues/3114#issuecomment-812179743) for a fix.

**Comment by IVerkhosh** [comment]
Hi Dustin,
The fix you referring me to fix the "Could not infer a metadata type" issue, but invoked another issue: ERROR running force:source:retrieve:  Unexpected file found in package directory. I fixed one bu adding audience to metadataTypeInfos.json file in the node_modules\salesforce-alm\metadata. But now it complaining about
*\force-app\main\default\customindex\Account.AccountNumber.indx-meta.xml.
Even if I delete custom index folder, it start complaining about objects.
How to deal with it?

**Comment by CaptainDX** [comment]
I got the same error with botVersion-meta.xml: Could not infer metadata type

**Comment by IVerkhosh** [comment]
Hi @CaptainDX,
My issues been fixed. I found the folder which causing the problem. In my case it was CustomIndex.
To find it, was a hard one. With Experimental: Deploy Retrieve turned OFF, I created a brand new SFDX Project with manifest. Run the Package Generator to pull all components in my ORG, and run sfdx force:source:retrieve -m "ActionLinkGroupTemplate,	AnimationRule,	ApexClass,	ApexComponent,	ApexEmailNotifications,	ApexPage,	ApexTestSuite,	ApexTrigger" in pieces to see where it breaks. And like this for all 148 components.
